app/Console/Commands/DummyEntities.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Location;
+use App\Models\Family;
+use App\Models\Note;
+use App\Models\Character;
+use App\Models\Organisation;
+use App\Models\Event;
+use App\Models\Item;
+use App\Models\Tag;
+use App\Models\Race;
+use App\Models\Campaign;
+use App\Observers\LocationObserver;
+use App\Observers\CharacterObserver;
+use App\Observers\FamilyObserver;
+use App\Observers\OrganisationObserver;
+use App\Observers\EventObserver;
+use App\Observers\ItemObserver;
+use App\Observers\TagObserver;
+use App\Observers\RaceObserver;
+use App\Observers\NoteObserver;
+use Illuminate\Support\Str;
+use App\Facades\CampaignLocalization;
+use Illuminate\Database\Eloquent\Factories\Sequence;
+use Illuminate\Console\Command;
+
+
+class DummyJobLog extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'campaign:populate {campaign}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Generate dummy entities in a specified campaign';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $campaignId = $this->argument('campaign');
+        $this->loadObservers();
+
+        //Generate Characters and Locations
+        Location::factory()
+            ->state(['campaign_id' => $campaignId, 'name' => 'Thaelia', 'slug' => Str::slug('Thaelia')])
```

Comment:
You no longer need to set the slug as that is now handled in the observers :-)
I also think you can skip the `campaign_id` set now that you have the observers, but could be wrong